### ISSUE TITLE
BuiltIn Library: add "Run Keyword If Keyword" variations

### ISSUE THREAD

**Issue by vkruoso** [robotframework/robotframework#2201]
When user keyword fails and a keyword teardown is implemented, it is necessary to add the following boilerplate code if the teardown wants to run keywords when the user keyword itself has failed.

``` robotframework
*** Keywords ***
Some Keyword
    Fail  Just fails
    [Teardown]  Some Keyword Teardown

Some Keyword Teardown
    Run Keyword If  '${KEYWORD STATUS}' == 'FAIL'
    ...   Log  'The Keyword Failed'
```

It would be nice to have `Run Keyword If Keyword Failed` and `Run Keyword If Keyword Passed` keywords to handle that case.

**Comment by pekkaklarck** [comment]
Sounds good to me. Have you already looked at implementation?

**Comment by vkruoso** [comment]
Not really. Was waiting on the feedback, but I won't be able to look into it in the short term tough. It does not sound hard as there is implementation on the test case level. Will try to give it a try this weekend.

**Comment by pekkaklarck** [comment]
Cool. There's still good time to get this into RF 3.0 that we ought to get ready by the end of the year. Next preview release is planned for next week or week after.

This is related to #2078 about making status variables and passed/failed keywords to be available also outside teardowns. The new proposed keywords could be used, for example, after a continuable failure. No need to take that into account at this point, though.

**Comment by pekkaklarck** [comment]
Have you @vkruoso had time to look at this?

**Comment by vkruoso** [comment]
Sorry @pekkaklarck I did not. Will try to take a look at this soon, but no say it for sure when it will be.

**Comment by pekkaklarck** [comment]
We are going to create RF 3.0 in few days and do not plan to do any more code changes to it. Depending on the implementation, this can be considered in 3.0.x or latest in 3.1.

**Comment by pekkaklarck** [comment]
We are staring with RF 3.0.1 development with a target to get it released in June. Do you have time and interest to get this one included? Or should we just close the whole issue?

**Comment by vkruoso** [comment]
I'll take a look at implementing this in until the end of next week. I think those keywords can be very useful. Sorry about the delay.

**Comment by pekkaklarck** [comment]
This is a bit too big change for a minor release. Targeting for 3.1.

**Comment by pekkaklarck** [comment]
RF 3.1 getting ready and need to make decision related to this issue.

Reasons I think this issue should still be included:
+ Functionality is consistent with existing `Run Keyword If Test Failed/Passed`.
+ We got a PR and on a quick look the PR looks pretty good.

Reason I think this issue should not be included:
+ I consider also `Run Keyword If Test Failed/Passed` kind of a hack and would prefer other solutions to the underlying problem in general. Possible solutions include:
   - Have separate functionality to run a keyword at the end of a test/keyword *only* if there's a failure. Similar to teardown but not executed if test/keyword passes. The last keyword of the  test/keyword can already be used to handle the case where the test/keyword passes.
   - Have separate functionality to handle errors when they occur. Could allow registering a callback keyword that gets the occurred error as an argument.
   - Add separate `TRY/EXCEPT` syntax for handling failures. Obviously a bigger change requiring parser changes, but also something that would be very useful in the RPA context.
+ RF 3.1 is very long overdue. Going through the PR and fixing possible issues will take time.

How do you @vkruoso feel about this?

**Comment by pekkaklarck** [comment]
No comments about the validity of this issue lately. Removing from 3.1 scope but keeping open at least for the time being.
